src/Product.jl

```diff
@@ -0,0 +1,85 @@
+using LinearAlgebra
+using Graphs
+
+struct Product <: AbstractAnsatz
+    tn::Ansatz
+end
+
+Ansatz(tn::Product) = tn.tn
+
+Base.copy(x::Product) = Product(copy(Ansatz(x)))
+Base.similar(x::Product) = Product(similar(Ansatz(x)))
+Base.zero(x::Product) = Product(zero(Ansatz(x)))
+
+function Product(arrays::AbstractArray{<:AbstractVector})
+    n = length(arrays)
+    gen = IndexCounter()
+    symbols = map(arrays) do _
+        nextindex!(gen)
+    end
+    _tensors = map(eachindex(arrays)) do i
+        Tensor(arrays[i], [symbols[i]])
+    end
+
+    sitemap = Dict(Site(i) => symbols[i] for i in eachindex(arrays))
+    qtn = Quantum(TensorNetwork(_tensors), sitemap)
+    graph = Graph(n)
+    mapping = BijectiveIdDict{Site,Int}(Pair{Site,Int}[site => i for (i, site) in enumerate(lanes(qtn))])
+    lattice = Lattice(mapping, graph)
+    ansatz = Ansatz(qtn, lattice)
+    return Product(ansatz)
+end
+
+function Product(arrays::AbstractArray{<:AbstractMatrix})
+    n = length(arrays)
+    gen = IndexCounter()
+    symbols = map(arrays) do _
+        (nextindex!(gen), nextindex!(gen))
+    end
+    _tensors = map(eachindex(arrays)) do i
+        Tensor(arrays[i], [symbols[i][1], symbols[i][2]])
+    end
+
+    sitemap = merge!(
+        Dict(Site(i; dual=true) => symbols[i][1] for i in eachindex(arrays)),
+        Dict(Site(i) => symbols[i][2] for i in eachindex(arrays)),
+    )
+    qtn = Quantum(TensorNetwork(_tensors), sitemap)
+    graph = Graph(n)
+    mapping = BijectiveIdDict{Site,Int}(Pair{Site,Int}[site => i for (i, site) in enumerate(lanes(qtn))])
+    lattice = Lattice(mapping, graph)
+    ansatz = Ansatz(qtn, lattice)
+    return Product(ansatz)
+end
+
+LinearAlgebra.norm(tn::Product, p::Real=2) = LinearAlgebra.norm(socket(tn), tn, p)
+function LinearAlgebra.norm(::Union{State,Operator}, tn::Product, p::Real)
+    return mapreduce(*, tensors(tn)) do tensor
+        norm(tensor, p)
+    end^(1//p)
+end
+
+LinearAlgebra.opnorm(tn::Product, p::Real=2) = LinearAlgebra.opnorm(socket(tn), tn, p)
+function LinearAlgebra.opnorm(::Operator, tn::Product, p::Real)
+    return mapreduce(*, tensors(tn)) do tensor
+        opnorm(parent(tensor), p)
+    end^(1//p)
+end
+
+LinearAlgebra.normalize!(tn::Product, p::Real=2) = LinearAlgebra.normalize!(socket(tn), tn, p)
```

Comment:
```suggestion
"""
    LinearAlgebra.normalize!(tn::Product, p::Real=2)

Normalize a [`Product`](@ref) ansatz.
"""
LinearAlgebra.normalize!(tn::Product, p::Real=2) = LinearAlgebra.normalize!(socket(tn), tn, p)
```

Reply:
mmm why do we need this docstring? it's not adding more information, it's obvious that `normalize!` will normalize the ansatz...

Reply:
Okay, so at least put the ones from `Product` please

Reply:
yes im on it